pgjdbc/src/main/java/org/postgresql/util/ReaderInputStream.java

```diff
@@ -0,0 +1,187 @@
+package org.postgresql.util;
+
+//#if mvn.project.property.postgresql.jdbc.spec >= "JDBC4.1"
+import static java.nio.charset.StandardCharsets.UTF_8;
+//#endif
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetEncoder;
+import java.nio.charset.CoderResult;
+import java.nio.charset.MalformedInputException;
+
+public class ReaderInputStream extends InputStream {
+  private static final int DEFAULT_CHAR_BUFFER_SIZE = 8 * 1024;
+
+  //#if mvn.project.property.postgresql.jdbc.spec < "JDBC4.1"
+  private static final Charset UTF_8 = Charset.availableCharsets().get("UTF-8");
```

Comment:
1. Charset.forName("UTF-8") throws an exception. Charset.availableCharsets().get("UTF-8") does not. Given that "UTF-8" is a mandatory charset, it doesn't make a huge difference, but some static analyzers complain if exceptions can be thrown during class initialization.

1. I couldn't find any discussion of how long pgjdbc intends to maintain Java 6 compatibility. Given that Java 6 has been out of support for years, I would consider dropping Java 6 support to be a better option. As a compromise, I coded it to use StandardCharset when available and fall back to a static final class member for Java 6.